crates/examples/src/readobj/macho.rs

```diff
@@ -36,23 +36,19 @@ pub(super) fn print_dyld_cache_header(
     });
 }
 
-pub(super) fn print_dyld_cache_mappings(
-    p: &mut Printer<'_>,
-    endian: Endianness,
-    mappings: &[DyldCacheMappingInfo<Endianness>],
-) {
+pub(super) fn print_dyld_cache_mappings(p: &mut Printer<'_>, mappings: &DyldCacheMappingSlice) {
     if !p.options.file {
         return;
     }
-    for mapping in mappings {
-        p.group("DyldCacheMappingInfo", |p| {
-            p.field_hex("Address", mapping.address.get(endian));
-            p.field_hex("Size", mapping.size.get(endian));
-            p.field_hex("FileOffset", mapping.file_offset.get(endian));
-            p.field_hex("MaxProt", mapping.max_prot.get(endian));
-            p.flags(mapping.max_prot.get(endian), 0, FLAGS_VM);
-            p.field_hex("InitProt", mapping.init_prot.get(endian));
-            p.flags(mapping.init_prot.get(endian), 0, FLAGS_VM);
+    for mapping in mappings.iter() {
+        p.group("DyldCacheMapping", |p| {
```

Comment:
These should stay as `DyldCacheMappingInfo` and we should handle `DyldCacheMappingAndSlideInfo` separately.